app/Http/Requests/V1/TimeEntry/TimeEntryAggregateRequest.php

```diff
@@ -0,0 +1,138 @@
+                'string',
+                'uuid',
+                new ExistsEloquent(Member::class, null, function (Builder $builder): Builder {
+                    /** @var Builder<Member> $builder */
+                    return $builder->whereBelongsTo($this->organization, 'organization');
+                }),
+            ],
+            // Filter by multiple member IDs, member IDs are OR combined, but AND combined with the member_id parameter
+            'member_ids' => [
+                'array',
+                'min:1',
+            ],
+            'member_ids.*' => [
+                'string',
+                'uuid',
+                new ExistsEloquent(Member::class, null, function (Builder $builder): Builder {
+                    /** @var Builder<Member> $builder */
+                    return $builder->whereBelongsTo($this->organization, 'organization');
+                }),
+            ],
+
+            // Filter by user ID
+            'user_id' => [
+                'string',
+                'uuid',
+                new ExistsEloquent(User::class, null, function (Builder $builder): Builder {
+                    /** @var Builder<User> $builder */
+                    return $builder->belongsToOrganization($this->organization);
+                }),
+            ],
+            // Filter by project IDs, project IDs are OR combined
+            'project_ids' => [
+                'array',
+                'min:1',
+            ],
+            'project_ids.*' => [
+                'string',
+                'uuid',
+                new ExistsEloquent(Project::class, null, function (Builder $builder): Builder {
+                    /** @var Builder<Project> $builder */
+                    return $builder->visibleByUser(Auth::user());
+                }),
+            ],
+            // Filter by tag IDs, tag IDs are AND combined
+            'tag_ids' => [
+                'array',
+                'min:1',
+            ],
+            'tag_ids.*' => [
+                'string',
+                'uuid',
+                new ExistsEloquent(Tag::class, null, function (Builder $builder): Builder {
+                    /** @var Builder<Tag> $builder */
+                    return $builder->whereBelongsTo($this->organization, 'organization');
+                }),
+            ],
+            // Filter by task IDs, task IDs are OR combined
+            'task_ids' => [
+                'array',
+                'min:1',
+            ],
+            'task_ids.*' => [
+                'string',
+                'uuid',
+                new ExistsEloquent(Task::class, null, function (Builder $builder): Builder {
+                    /** @var Builder<Task> $builder */
+                    return $builder->visibleByUser(Auth::user());
+                }),
+            ],
+            // Filter only time entries that have a start date before (not including) the given date (example: 2021-12-31)
+            'before' => [
```

Comment:
@korridor it might be better to do start and end or start_date and end_date here. not including the given day is quite confusing IMO. usually you'll have some sort of range you want to have the data for, but this way you would always have to calculate +1 and -1 on the ends. 

oh and the validation rule is not matching the description here.

Reply:
ah ok i see i thought the validation rule is wrong but its probably the comment that is wrong. i think before and after is more difficult to parse than start and end, but the +1 -1 problem is not relevant with a timestamp i guess.

Reply:
I just corrected the comment to correctly reflect the current behaviour. Your suggestion is now to rename it to `start` and `end` instead of `before` and `after`?

Reply:
`start` would become `after`, and the `end` would become `before`, yes. I just think it is confusing to think about a time range as before and after, because before is the end, and after is the beginning. i would guess the validation rule was the wrong way around for that reason as well. and i also did it the wrong was around initially on the client.